go/vt/workflow/node.go

```diff
@@ -94,13 +94,17 @@ type Node struct {
 	// workflow is the workflow which owns this node.
 	workflow Workflow
 
-	// insideModify is a marker set by the Modify method.
-	// It is checked by methods that need to be called inside Modify.
-	insideModify bool
-
 	// Next are all the attributes that are exported to node.ts.
+	// Note that even though Path is publicly accessible it should be
+	// changed only by Node and NodeManager, otherwise things can break.
+	// It's public only to be exportable to json. Similarly PathName (which
+	// is the last component of path) on top-level node should be populated
+	// only by NodeManager. For non-top-level children it can be changed
+	// at any time, Path will be adjusted correspondingly when the parent
+	// node's BroadcastChanges() is called.
 
 	Name            string                   `json:"name"`
+	PathName        string                   `json:"pathName"`
 	Path            string                   `json:"path"`
 	Children        []*Node                  `json:"children"`
 	LastChanged     int64                    `json:"lastChanged"`
```

Comment:
LastChanged is not updated correctly it seems. It's only set when we add a root node.